app/src/main/java/tech/relaycorp/gateway/background/endpoint/EndpointPreRegistrationService.kt

```diff
@@ -0,0 +1,55 @@
+package tech.relaycorp.gateway.background.endpoint
+
+import android.app.Service
+import android.content.Intent
+import android.os.Handler
+import android.os.IBinder
+import android.os.Looper
+import android.os.Message
+import android.os.Messenger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.runBlocking
+import tech.relaycorp.gateway.background.component
+import tech.relaycorp.gateway.domain.endpoint.EndpointRegistration
+import javax.inject.Inject
+
+class EndpointPreRegistrationService : Service() {
+    private val scope get() = CoroutineScope(Dispatchers.IO)
+
+    @Inject
+    lateinit var endpointRegistration: EndpointRegistration
+
+    override fun onBind(intent: Intent): IBinder? {
+        component.inject(this)
+        val messenger = Messenger(
+            object : Handler(Looper.getMainLooper()) {
+                override fun handleMessage(msg: Message) {
+                    when (msg.what) {
+                        PREREGISTRATION_REQUEST -> reply(msg)
+                    }
+                }
+            }
+        )
+        return messenger.binder
+    }
+
+    // TODO: Replace runBlocking with scope.launch
+    internal fun reply(requestMessage: Message) = runBlocking {
+        // TODO: Dynamically compute the application id
+        val craSerialized = endpointRegistration.authorize("temp-app-id")
```

Comment:
This is another thing that was breaking in the instrumentation test: I'd like to "discover" the `applicationId` (see: https://stackoverflow.com/q/12918731/129437), but I kept getting the current app's id. I guess that's because of the context in which the test is run, right?

Presumably we'll have to mock this in the test, but I wanted to check I'm not missing something.